docs/man/man1/kubectl.1
.TH "KUBERNETES" "1" " kubernetes User Manuals" "Eric Paris" "Jan 2015"  ""


.SH NAME
.PP
kubectl \- kubectl controls the Kubernetes cluster manager


.SH SYNOPSIS
.PP
\fBkubectl\fP [OPTIONS]


.SH DESCRIPTION
.PP
kubectl controls the Kubernetes cluster manager.

.PP
Find more information at 
\[la]https://github.com/kubernetes/kubernetes\[ra].


.SH OPTIONS
.PP
\fB\-\-allow\-verification\-with\-non\-compliant\-keys\fP=false
    Allow a SignatureVerifier to use keys which are technically non\-compliant with RFC6962.

.PP
\fB\-\-alsologtostderr\fP=false
    log to standard error as well as files

.PP
\fB\-\-application\-metrics\-count\-limit\fP=100
    Max number of application metrics to store (per container)

.PP
\fB\-\-as\fP=""
    Username to impersonate for the operation

.PP
\fB\-\-as\-group\fP=[]
    Group to impersonate for the operation, this flag can be repeated to specify multiple groups.

.PP
\fB\-\-azure\-container\-registry\-config\fP=""
    Path to the file container Azure container registry configuration information.

.PP
\fB\-\-boot\-id\-file\fP="/proc/sys/kernel/random/boot\_id"
    Comma\-separated list of files to check for boot\-id. Use the first one that exists.

.PP
\fB\-\-cache\-dir\fP="/home/username/.kube/http\-cache"
    Default HTTP cache directory

.PP
\fB\-\-certificate\-authority\fP=""
    Path to a cert file for the certificate authority

.PP
\fB\-\-client\-certificate\fP=""
    Path to a client certificate file for TLS

.PP
\fB\-\-client\-key\fP=""
    Path to a client key file for TLS

.PP
\fB\-\-cloud\-provider\-gce\-lb\-src\-cidrs\fP=209.85.204.0/22,130.211.0.0/22,35.191.0.0/16,209.85.152.0/22
    CIDRs opened in GCE firewall for LB traffic proxy \& health checks

.PP
\fB\-\-cluster\fP=""
    The name of the kubeconfig cluster to use

.PP
\fB\-\-container\-hints\fP="/etc/cadvisor/container\_hints.json"
    location of the container hints file

.PP
\fB\-\-containerd\fP="unix:///var/run/containerd.sock"
    containerd endpoint

.PP
\fB\-\-context\fP=""
    The name of the kubeconfig context to use

.PP
\fB\-\-default\-not\-ready\-toleration\-seconds\fP=300
    Indicates the tolerationSeconds of the toleration for notReady:NoExecute that is added by default to every pod that does not already have such a toleration.

.PP
\fB\-\-default\-unreachable\-toleration\-seconds\fP=300
    Indicates the tolerationSeconds of the toleration for unreachable:NoExecute that is added by default to every pod that does not already have such a toleration.

.PP
\fB\-\-docker\fP="unix:///var/run/docker.sock"
    docker endpoint

.PP
\fB\-\-docker\-env\-metadata\-whitelist\fP=""
    a comma\-separated list of environment variable keys that needs to be collected for docker containers

.PP
\fB\-\-docker\-only\fP=false
    Only report docker containers in addition to root stats

.PP
\fB\-\-docker\-root\fP="/var/lib/docker"
    DEPRECATED: docker root is read from docker info (this is a fallback, default: /var/lib/docker)

.PP
\fB\-\-docker\-tls\fP=false
    use TLS to connect to docker

.PP
\fB\-\-docker\-tls\-ca\fP="ca.pem"
    path to trusted CA

.PP
\fB\-\-docker\-tls\-cert\fP="cert.pem"
    path to client certificate

.PP
\fB\-\-docker\-tls\-key\fP="key.pem"
    path to private key

.PP
\fB\-\-enable\-load\-reader\fP=false
    Whether to enable cpu load reader

.PP
\fB\-\-event\-storage\-age\-limit\fP="default=0"
    Max length of time for which to store events (per type). Value is a comma separated list of key values, where the keys are event types (e.g.: creation, oom) or "default" and the value is a duration. Default is applied to all non\-specified event types

.PP
\fB\-\-event\-storage\-event\-limit\fP="default=0"
    Max number of events to store (per type). Value is a comma separated list of key values, where the keys are event types (e.g.: creation, oom) or "default" and the value is an integer. Default is applied to all non\-specified event types

.PP
\fB\-\-global\-housekeeping\-interval\fP=1m0s
    Interval between global housekeepings

.PP
\fB\-\-google\-json\-key\fP=""
    The Google Cloud Platform Service Account JSON Key to use for authentication.

.PP
\fB\-\-housekeeping\-interval\fP=10s
    Interval between container housekeepings

.PP
\fB\-\-insecure\-skip\-tls\-verify\fP=false
    If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure

.PP
\fB\-\-ir\-data\-source\fP="influxdb"
    Data source used by InitialResources. Supported options: influxdb, gcm.

.PP
\fB\-\-ir\-dbname\fP="k8s"
    InfluxDB database name which contains metrics required by InitialResources

.PP
\fB\-\-ir\-hawkular\fP=""
    Hawkular configuration URL

.PP
\fB\-\-ir\-influxdb\-host\fP="localhost:8080/api/v1/namespaces/kube\-system/services/monitoring\-influxdb:api/proxy"
    Address of InfluxDB which contains metrics required by InitialResources

.PP
\fB\-\-ir\-namespace\-only\fP=false
    Whether the estimation should be made only based on data from the same namespace.

.PP
\fB\-\-ir\-password\fP="root"
    Password used for connecting to InfluxDB

.PP
\fB\-\-ir\-percentile\fP=90
    Which percentile of samples should InitialResources use when estimating resources. For experiment purposes.

.PP
\fB\-\-ir\-user\fP="root"
    User used for connecting to InfluxDB

.PP
\fB\-\-kubeconfig\fP=""
    Path to the kubeconfig file to use for CLI requests.

.PP
\fB\-\-log\-backtrace\-at\fP=:0
    when logging hits line file:N, emit a stack trace

.PP
\fB\-\-log\-cadvisor\-usage\fP=false
    Whether to log the usage of the cAdvisor container

.PP
\fB\-\-log\-dir\fP=""
    If non\-empty, write log files in this directory

.PP
\fB\-\-loglevel\fP=1
    Log level (0 = DEBUG, 5 = FATAL)

.PP
\fB\-\-logtostderr\fP=false
    log to standard error instead of files

.PP
\fB\-\-machine\-id\-file\fP="/etc/machine\-id,/var/lib/dbus/machine\-id"
    Comma\-separated list of files to check for machine\-id. Use the first one that exists.

.PP
\fB\-\-match\-server\-version\fP=false
    Require server version to match client version

.PP
\fB\-n\fP, \fB\-\-namespace\fP=""
    If present, the namespace scope for this CLI request

.PP
\fB\-\-password\fP=""
    Password for basic authentication to the API server

.PP
\fB\-\-request\-timeout\fP="0"
    The length of time to wait before giving up on a single server request. Non\-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests.

.PP
\fB\-s\fP, \fB\-\-server\fP=""
    The address and port of the Kubernetes API server

.PP
\fB\-\-stderrthreshold\fP=2
    logs at or above this threshold go to stderr

.PP
\fB\-\-storage\-driver\-buffer\-duration\fP=1m0s
    Writes in the storage driver will be buffered for this duration, and committed to the non memory backends as a single transaction

.PP
\fB\-\-storage\-driver\-db\fP="cadvisor"
    database name

.PP
\fB\-\-storage\-driver\-host\fP="localhost:8086"
    database host:port

.PP
\fB\-\-storage\-driver\-password\fP="root"
    database password

.PP
\fB\-\-storage\-driver\-secure\fP=false
    use secure connection with database

.PP
\fB\-\-storage\-driver\-table\fP="stats"
    table name

.PP
\fB\-\-storage\-driver\-user\fP="root"
    database username

.PP
\fB\-\-token\fP=""
    Bearer token for authentication to the API server

.PP
\fB\-\-user\fP=""
    The name of the kubeconfig user to use

.PP
\fB\-\-username\fP=""
    Username for basic authentication to the API server

.PP
\fB\-v\fP, \fB\-\-v\fP=0
    log level for V logs

.PP
\fB\-\-version\fP=false
    Print version information and quit

.PP
\fB\-\-vmodule\fP=
    comma\-separated list of pattern=N settings for file\-filtered logging


.SH SEE ALSO
.PP
\fBkubectl\-alpha(1)\fP, \fBkubectl\-annotate(1)\fP, \fBkubectl\-api\-versions(1)\fP, \fBkubectl\-apply(1)\fP, \fBkubectl\-attach(1)\fP, \fBkubectl\-auth(1)\fP, \fBkubectl\-autoscale(1)\fP, \fBkubectl\-certificate(1)\fP, \fBkubectl\-cluster\-info(1)\fP, \fBkubectl\-completion(1)\fP, \fBkubectl\-config(1)\fP, \fBkubectl\-convert(1)\fP, \fBkubectl\-cordon(1)\fP, \fBkubectl\-cp(1)\fP, \fBkubectl\-create(1)\fP, \fBkubectl\-delete(1)\fP, \fBkubectl\-describe(1)\fP, \fBkubectl\-drain(1)\fP, \fBkubectl\-edit(1)\fP, \fBkubectl\-exec(1)\fP, \fBkubectl\-explain(1)\fP, \fBkubectl\-expose(1)\fP, \fBkubectl\-get(1)\fP, \fBkubectl\-label(1)\fP, \fBkubectl\-logs(1)\fP, \fBkubectl\-options(1)\fP, \fBkubectl\-patch(1)\fP, \fBkubectl\-plugin(1)\fP, \fBkubectl\-port\-forward(1)\fP, \fBkubectl\-proxy(1)\fP, \fBkubectl\-replace(1)\fP, \fBkubectl\-rolling\-update(1)\fP, \fBkubectl\-rollout(1)\fP, \fBkubectl\-run(1)\fP, \fBkubectl\-run\-container(1)\fP, \fBkubectl\-scale(1)\fP, \fBkubectl\-set(1)\fP, \fBkubectl\-taint(1)\fP, \fBkubectl\-top(1)\fP, \fBkubectl\-uncordon(1)\fP, \fBkubectl\-version(1)\fP,


.SH HISTORY
.PP
January 2015, Originally compiled by Eric Paris (eparis at redhat dot com) based on the kubernetes source material, but hopefully they have been automatically generated since!


news.md
---
layout: news
title: News
permalink: /news/
show_in_nav: true
slug: news

hero:
    title: News
    text: Read what others are saying about Kubernetes.
    img: /img/desktop/news/hero_icon.svg

default_img: /img/desktop/news/img_placeholder@2x.jpg

load_more_label: Load More

---


cluster/kubectl.sh
#!/bin/bash

# Copyright 2014 The Kubernetes Authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

set -o errexit
set -o nounset
set -o pipefail

# Stop the bleeding, turn off the warning until we fix token gen.
# echo "-=-=-=-=-=-=-=-=-=-="
# echo "NOTE:"
# echo "kubectl.sh is deprecated and will be removed soon."
# echo "please replace all usage with calls to the kubectl"
# echo "binary and ensure that it is in your PATH." 
# echo ""
# echo "Please see 'kubectl help config' for more details"
# echo "about configuring kubectl for your cluster."
# echo "-=-=-=-=-=-=-=-=-=-="


KUBE_ROOT=${KUBE_ROOT:-$(dirname "${BASH_SOURCE}")/..}
source "${KUBE_ROOT}/cluster/kube-util.sh"
source "${KUBE_ROOT}/cluster/clientbin.sh"

# If KUBECTL_PATH isn't set, gather up the list of likely places and use ls
# to find the latest one.
if [[ -z "${KUBECTL_PATH:-}" ]]; then
  kubectl=$( get_bin "kubectl" "cmd/kubectl" )

  if [[ ! -x "$kubectl" ]]; then
    print_error "kubectl"
    exit 1
  fi
elif [[ ! -x "${KUBECTL_PATH}" ]]; then
  {
    echo "KUBECTL_PATH environment variable set to '${KUBECTL_PATH}', but "
    echo "this doesn't seem to be a valid executable."
  } >&2
  exit 1
fi
kubectl="${KUBECTL_PATH:-${kubectl}}"

if [[ "$KUBERNETES_PROVIDER" == "gke" ]]; then
  detect-project &> /dev/null
elif [[ "$KUBERNETES_PROVIDER" == "ubuntu" ]]; then
  detect-master > /dev/null
  config=(
    "--server=http://${KUBE_MASTER_IP}:8080"
  )
fi


if false; then
  # disable these debugging messages by default
  echo "current-context: \"$(${kubectl} "${config[@]:+${config[@]}}" config view -o template --template='{{index . "current-context"}}')\"" >&2
  echo "Running:" "${kubectl}" "${config[@]:+${config[@]}}" "${@+$@}" >&2
fi

if [[ "${1:-}" =~ ^(path)$ ]]; then
  echo "${kubectl}"
  exit 0
fi

"${kubectl}" "${config[@]:+${config[@]}}" "${@+$@}"

